### ISSUE TITLE
Paths in package recipes used without processing variables

### ISSUE THREAD

**Issue by s-ludwig** [D-Programming-Language/dub#784]
Quote from http://forum.rejectedsoftware.com/groups/rejectedsoftware.dub/thread/5915/

> I can use $HOME in statements such as:
> 
> lflags "-L$HOME/Built/lib/gtkd-head-dmd/lib" "-lgtkd-3" "-lgstreamerd-3" platform="dmd"
> 
> and everything is hunky dory. If I use $HOME is statemetns such as:
> 
> importPaths "$HOME/Built/lib/gtkd-head-dmd/include/d/gtkd-3/" platform="dmd"
> 
> the build works fine but I keep getting messages:
> 
> Invalid source/import path: /home/users/russel/Repositories/Git/Git/Foscam/$HOME/Built/lib/gtkd-head-dmd/include/d/gtkd-3/
> 
> but, as I say, the build actually works fine. However, when I start Emacs with the ac-dcd package installed and used I get the message:
> 
> "dcd-client -I/usr/include/dmd/phobos -I/usr/include/dmd/druntime/import -I/home/users/russel/Repositories/Git/Git/Foscam/source/ -I/home/users/russel/Repositories/Git/Git/Foscam/$HOME/Built/lib/gtkd-head-dmd/include/d/gtkd-3/" failed.
> Error type is: Unable to connect socket : No such file or directory
> 
> Which is believe is attributable to the string dub returns being wrong.

`Package.getPlatformSettings` gets invoked on a package instance that doesn't yet has its variables substituted using `Project.processVars`. Maybe the API should be changed to shift variable substitution to this place to avoid similar mistakes in the future.

**Comment by jeanbaptistelab** [comment]
@s-ludwig I believe the issue mentioned by @russel in this post is not due to the lack variable expansion, but rather by a recent change in DCD's command line interface that now seems to mandate the use of the "--tcp" command line option whenever the "--port" option is used. I've fixed this issue in ac-dcd (which is where it really is) in my fork here: https://github.com/jeanbaptistelab/ac-dcd

Cheers,

JB

**Comment by s-ludwig** [comment]
Good point, I kind of skipped over that part a bit. But there is definitely a variable expansion issue on top of that.
